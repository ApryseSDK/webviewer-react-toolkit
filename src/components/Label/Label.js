'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Label = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importStar(require('react'));
var useID_1 = require('../../hooks/useID');
var Label = function(_a) {
  var label = _a.label,
    optionalText = _a.optionalText,
    children = _a.children,
    className = _a.className,
    htmlFor = _a.htmlFor,
    props = tslib_1.__rest(_a, [
      'label',
      'optionalText',
      'children',
      'className',
      'htmlFor',
    ]);
  var childrenId =
    children === null || children === void 0 ? void 0 : children.props.id;
  var id = (0, useID_1.useID)(childrenId);
  var labelClass = (0, classnames_1.default)(
    'ui__base ui__label',
    {
      'ui__label--disabled':
        children === null || children === void 0
          ? void 0
          : children.props.disabled,
      'ui__label--attached': children,
    },
    className,
  );
  return react_1.default.createElement(
    react_1.default.Fragment,
    null,
    react_1.default.createElement(
      'label',
      tslib_1.__assign({}, props, {
        className: labelClass,
        htmlFor: htmlFor !== null && htmlFor !== void 0 ? htmlFor : id,
      }),
      label,
      optionalText
        ? react_1.default.createElement(
            'span',
            { className: 'ui__label__optional' },
            optionalText,
          )
        : undefined,
    ),
    children ? (0, react_1.cloneElement)(children, { id: id }) : undefined,
  );
};
exports.Label = Label;
//# sourceMappingURL=Label.js.map
