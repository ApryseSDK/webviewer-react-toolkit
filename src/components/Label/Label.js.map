{"version":3,"file":"Label.js","sourceRoot":"","sources":["Label.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,qDAA8F;AAC9F,2CAA0C;AAoBnC,IAAM,KAAK,GAAmB,UAAC,EAA+D;IAA7D,IAAA,KAAK,WAAA,EAAE,YAAY,kBAAA,EAAE,QAAQ,cAAA,EAAE,SAAS,eAAA,EAAE,OAAO,aAAA,EAAK,KAAK,sBAA7D,6DAA+D,CAAF;IACjG,IAAM,UAAU,GAAG,QAAQ,aAAR,QAAQ,uBAAR,QAAQ,CAAE,KAAK,CAAC,EAAE,CAAC;IACtC,IAAM,EAAE,GAAG,IAAA,aAAK,EAAC,UAAU,CAAC,CAAC;IAE7B,IAAM,UAAU,GAAG,IAAA,oBAAU,EAC3B,oBAAoB,EACpB,EAAE,qBAAqB,EAAE,QAAQ,aAAR,QAAQ,uBAAR,QAAQ,CAAE,KAAK,CAAC,QAAQ,EAAE,qBAAqB,EAAE,QAAQ,EAAE,EACpF,SAAS,CACV,CAAC;IAEF,OAAO,CACL;QACE,4DAAW,KAAK,IAAE,SAAS,EAAE,UAAU,EAAE,OAAO,EAAE,OAAO,aAAP,OAAO,cAAP,OAAO,GAAI,EAAE;YAC5D,KAAK;YACL,YAAY,CAAC,CAAC,CAAC,wCAAM,SAAS,EAAC,qBAAqB,IAAE,YAAY,CAAQ,CAAC,CAAC,CAAC,SAAS,CACjF;QACP,QAAQ,CAAC,CAAC,CAAC,IAAA,oBAAY,EAAC,QAAQ,EAAE,EAAE,EAAE,IAAA,EAAE,CAAC,CAAC,CAAC,CAAC,SAAS,CACrD,CACJ,CAAC;AACJ,CAAC,CAAC;AAnBW,QAAA,KAAK,SAmBhB"}