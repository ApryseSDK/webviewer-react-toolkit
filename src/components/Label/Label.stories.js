'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Detached = exports.WithDisabledFormElement = exports.WithCustomId = exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var Input_1 = require('../Input');
var Label_1 = require('../Label');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Label',
  component: Label_1.Label,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(
    Label_1.Label,
    {
      label: (0, addon_knobs_1.text)('label', 'Labeled input field'),
      optionalText: (0, addon_knobs_1.boolean)('has optionalText?', false)
        ? '- optional'
        : undefined,
    },
    react_1.default.createElement(Input_1.Input, null),
  );
};
exports.Basic = Basic;
var WithCustomId = function() {
  return react_1.default.createElement(
    Label_1.Label,
    {
      label: (0, addon_knobs_1.text)('label', 'Labeled input field'),
      optionalText: (0, addon_knobs_1.boolean)('has optionalText?', false)
        ? '- optional'
        : undefined,
    },
    react_1.default.createElement(Input_1.Input, { id: 'custom_input_id' }),
  );
};
exports.WithCustomId = WithCustomId;
var WithDisabledFormElement = function() {
  return react_1.default.createElement(
    Label_1.Label,
    {
      label: (0, addon_knobs_1.text)('label', 'Labeled input field'),
      optionalText: (0, addon_knobs_1.boolean)('has optionalText?', false)
        ? '- optional'
        : undefined,
    },
    react_1.default.createElement(Input_1.Input, {
      id: 'custom_input_id',
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
    }),
  );
};
exports.WithDisabledFormElement = WithDisabledFormElement;
var Detached = function() {
  return react_1.default.createElement(
    react_1.default.Fragment,
    null,
    react_1.default.createElement(Label_1.Label, {
      htmlFor: 'custom_input_id',
      label: (0, addon_knobs_1.text)('label', 'Labeled input field'),
      optionalText: (0, addon_knobs_1.boolean)('has optionalText?', false)
        ? '- optional'
        : undefined,
    }),
    react_1.default.createElement('p', null, 'Some other stuff'),
    react_1.default.createElement(Input_1.Input, {
      id: 'custom_input_id',
      disabled: (0, addon_knobs_1.boolean)('disabled', false),
    }),
  );
};
exports.Detached = Detached;
//# sourceMappingURL=Label.stories.js.map
