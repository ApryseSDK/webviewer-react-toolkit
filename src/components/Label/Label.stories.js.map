{"version":3,"file":"Label.stories.js","sourceRoot":"","sources":["Label.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAuD;AACvD,wDAA0B;AAC1B,kCAAiC;AACjC,kCAAiC;AACjC,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,kBAAkB,EAAE,SAAS,EAAE,aAAK,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAEhF,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,aAAK,IACJ,KAAK,EAAE,IAAA,kBAAI,EAAC,OAAO,EAAE,qBAAqB,CAAC,EAC3C,YAAY,EAAE,IAAA,qBAAO,EAAC,mBAAmB,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,SAAS;IAE5E,8BAAC,aAAK,OAAG,CACH,CACT,EAP0B,CAO1B,CAAC;AAPW,QAAA,KAAK,SAOhB;AAEK,IAAM,YAAY,GAAG,cAAM,OAAA,CAChC,8BAAC,aAAK,IACJ,KAAK,EAAE,IAAA,kBAAI,EAAC,OAAO,EAAE,qBAAqB,CAAC,EAC3C,YAAY,EAAE,IAAA,qBAAO,EAAC,mBAAmB,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,SAAS;IAE5E,8BAAC,aAAK,IAAC,EAAE,EAAC,iBAAiB,GAAG,CACxB,CACT,EAPiC,CAOjC,CAAC;AAPW,QAAA,YAAY,gBAOvB;AAEK,IAAM,uBAAuB,GAAG,cAAM,OAAA,CAC3C,8BAAC,aAAK,IACJ,KAAK,EAAE,IAAA,kBAAI,EAAC,OAAO,EAAE,qBAAqB,CAAC,EAC3C,YAAY,EAAE,IAAA,qBAAO,EAAC,mBAAmB,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,SAAS;IAE5E,8BAAC,aAAK,IAAC,EAAE,EAAC,iBAAiB,EAAC,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,GAAI,CAC9D,CACT,EAP4C,CAO5C,CAAC;AAPW,QAAA,uBAAuB,2BAOlC;AAEK,IAAM,QAAQ,GAAG,cAAM,OAAA,CAC5B;IACE,8BAAC,aAAK,IACJ,OAAO,EAAC,iBAAiB,EACzB,KAAK,EAAE,IAAA,kBAAI,EAAC,OAAO,EAAE,qBAAqB,CAAC,EAC3C,YAAY,EAAE,IAAA,qBAAO,EAAC,mBAAmB,EAAE,KAAK,CAAC,CAAC,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,SAAS,GAC5E;IACF,4DAAuB;IACvB,8BAAC,aAAK,IAAC,EAAE,EAAC,iBAAiB,EAAC,QAAQ,EAAE,IAAA,qBAAO,EAAC,UAAU,EAAE,KAAK,CAAC,GAAI,CACnE,CACJ,EAV6B,CAU7B,CAAC;AAVW,QAAA,QAAQ,YAUnB"}