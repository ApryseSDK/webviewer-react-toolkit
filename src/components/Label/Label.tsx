import classnames from 'classnames';
import React, { cloneElement, FC, LabelHTMLAttributes, ReactElement, ReactNode, useMemo } from 'react';
import { getStringId } from '../../utils';

export interface LabelProps extends LabelHTMLAttributes<HTMLLabelElement> {
  /**
   * The label to apply to any form element.
   */
  label: ReactNode;
  /**
   * Pass a child element which can accept an `id` prop. If you don't specify
   * the `id` prop, one will be generated to link the label to the element. If
   * you wish to link this label to a form field without passing children, you
   * should specify the `htmlFor` prop with the `id` of your form field.
   */
  children?: ReactElement;
}

export const Label: FC<LabelProps> = ({ label, children, className, htmlFor, ...props }) => {
  const childrenId = children?.props.id;
  const id = useMemo(() => {
    if (childrenId) return childrenId;
    return getStringId('label');
  }, [childrenId]);

  const labelClass = classnames(
    'ui__base ui__label',
    { 'ui__label--disabled': children?.props.disabled, 'ui__label--attached': children },
    className,
  );

  return (
    <>
      <label {...props} className={labelClass} htmlFor={htmlFor ?? id}>
        {label}
      </label>
      {children ? cloneElement(children, { id }) : undefined}
    </>
  );
};
