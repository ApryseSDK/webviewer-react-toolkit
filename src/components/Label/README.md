A generic label for labelling form fields. If the child form field has an `id`
specified, that will be automatically parsed and used in the label's `htmlFor`
attribute to link the two. If no `id` is specified for the child, one will be
automatically generated.

You can also link the label to a form field without passing it as a child by
giving the field an `id` and passing the same `id` to the `htmlFor` prop for the
input. This is useful if the two elements do not sit near each other on the
page.

If the child element is disabled, the label will automatically change appearance
to a disabled state. This will not happen if the form field is not the child.
