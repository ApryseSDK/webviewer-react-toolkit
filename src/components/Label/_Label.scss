.ui__label {
  &--attached::after {
    height: 2px;
    width: 1px;
    content: '';
    display: block;
  }

  &--disabled {
    opacity: 0.7;
  }
}
