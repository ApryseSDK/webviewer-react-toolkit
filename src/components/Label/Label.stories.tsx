import { boolean, text } from '@storybook/addon-knobs';
import React from 'react';
import { Input } from '../Input';
import { Label } from '../Label';
import readme from './README.md';

export default { title: 'Components/Label', component: Label, parameters: { readme } };

export const Basic = () => (
  <Label
    label={text('label', 'Labeled input field')}
    optionalText={boolean('has optionalText?', false) ? '- optional' : undefined}
  >
    <Input />
  </Label>
);

export const WithCustomId = () => (
  <Label
    label={text('label', 'Labeled input field')}
    optionalText={boolean('has optionalText?', false) ? '- optional' : undefined}
  >
    <Input id="custom_input_id" />
  </Label>
);

export const WithDisabledFormElement = () => (
  <Label
    label={text('label', 'Labeled input field')}
    optionalText={boolean('has optionalText?', false) ? '- optional' : undefined}
  >
    <Input id="custom_input_id" disabled={boolean('disabled', false)} />
  </Label>
);

export const Detached = () => (
  <>
    <Label
      htmlFor="custom_input_id"
      label={text('label', 'Labeled input field')}
      optionalText={boolean('has optionalText?', false) ? '- optional' : undefined}
    />
    <p>Some other stuff</p>
    <Input id="custom_input_id" disabled={boolean('disabled', false)} />
  </>
);
