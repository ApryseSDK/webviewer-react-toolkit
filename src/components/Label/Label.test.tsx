import { shallow } from 'enzyme';
import React from 'react';
import { Label } from '../Label';

describe('Label component', () => {
  it('renders its contents', () => {
    const label = shallow(<Label label="" />);
    expect(label.find('.ui__label')).toHaveLength(1);
  });

  it('snapshot renders default label', () => {
    const label = shallow(<Label label="" />);
    expect(label).toMatchSnapshot();
  });
});
