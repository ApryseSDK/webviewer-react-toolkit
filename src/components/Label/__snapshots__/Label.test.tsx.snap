// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component snapshot renders default label 1`] = `
<Fragment>
  <label
    className="ui__base ui__label"
    htmlFor="label_1"
  />
</Fragment>
`;
